### PR TITLE
fix: assert deployment version length

### PR DESCRIPTION
**Pull request by akhilkumarpilli**
Fixes: #749

### CODE REVIEW

**Review by boz — CHANGES_REQUESTED** [review]
Looks great.

Can you define a constant in the file that generates the hash and use that constant here?

https://github.com/ovrclk/akash/blob/6ee9e6f06fc022595b2e0ced400c61b46f1aa72a/sdl/sdl.go#L120-L134

**Comment by akhilkumarpilli** [comment]
> Can you define a constant in the file that generates the hash and use that constant here?

Sorry, I didn't get it.

**Comment by hydrogen18** [comment]
I think @boz is suggesting you add a constant like `const ManifestVersionLength = 32` in `sdl/sdl.go` and then do `if len(version) != sdl.ManifestVersionLength` in the validation code.

**Comment by akhilkumarpilli** [comment]
@boz , Should I use this code to generate hash and use that constant in `ManifestVersion`?

https://github.com/ovrclk/akash/blob/master/testutil/ids.go#L41-L47

**Review by troian — APPROVED** [review]

File: x/deployment/types/types.go [hunk 1]
```diff
@@ -12,13 +12,18 @@ import (
 
 type attributesMatching map[string]types.Attributes
 
-// DefaultOrderBiddingDuration is the default time limit for an Order being active.
-// After the duration, the Order is automatically closed.
-// ( 24(hr) * 3600(seconds per hour) ) / 7s-Block
-const DefaultOrderBiddingDuration = int64(12342)
+const (
+	// ManifestVersionLength is the length of manifest version
+	ManifestVersionLength = 32
```
Comment (boz) [inline x/deployment/types/types.go:17]: It'd be nice to define this in the same place that the hash is calculated - `sdl/sdl.go`.  Are there import cycle issues if you do this?

If nothing else, you can use `sha256.Size`

Comment (akhilkumarpilli) [reply x/deployment/types/types.go:17]: Yes, there are import cycle issues if we add in `sdl/sdl.go`. That's the reason, I moved it to types.

**Review by boz — APPROVED** [review]
:ship:

**Pull request merged**